File: 프로그래머스/unrated/181871. 문자열이 몇 번 등장하는지 세기/문자열이 몇 번 등장하는지 세기.java
//1. 일반 풀이
// class Solution {
//     public int solution(String myString, String pat) {
//          int answer = 0;
//         for(int i = 0; i <= myString.length()-pat.length();i++){
//             if(myString.substring(i,i+pat.length()).equals(pat)) answer++;
//         }
//         return answer;
//     }
// }

//2. 스트림 풀이
import java.util.stream.IntStream;
class Solution {
    public int solution(String myString, String pat) {
        return (int) IntStream.rangeClosed(0, myString.length()-pat.length()).filter(i->myString.substring(i,i+pat.length()).equals(pat)).count();//카운트는 default가 long
    }
}

File: 프로그래머스/0/181871. 문자열이 몇 번 등장하는지 세기/문자열이 몇 번 등장하는지 세기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(string myString, string pat) {
    int answer = 0;
    size_t found = myString.find(pat);
    while(found != string::npos){
        answer++;
        found = myString.find(pat, found + 1);
    }
    return answer;
}